lib/vrl/tests/tests/expressions/if_statement/error_predicate.vrl

```diff
@@ -0,0 +1,32 @@
+# result: 
+# error[E111]: fallible predicate
+#   ┌─ :2:4
+#   │
+# 2 │ if contains(.noog, "thing") {
+#   │    ^^^^^^^^^^^^^^^^^^^^^^^^
+#   │    │
+#   │    this predicate can result in runtime error
+#   │    handle the error case to ensure runtime success
+#   │
+#   = see documentation about error handling at https://errors.vrl.dev/#handling
+#   = see language documentation at https://vrl.dev
+# 
+# error[E111]: fallible predicate
+#   ┌─ :6:4
+#   │
+# 6 │ if 23 && 32 {
+#   │    ^^^^^^^^
+#   │    │
+#   │    this predicate can result in runtime error
+#   │    handle the error case to ensure runtime success
+#   │
+#   = see documentation about error handling at https://errors.vrl.dev/#handling
+#   = see language documentation at https://vrl.dev
+
+if contains(.noog, "thing") {
+  log("ooh")
+}
+
+if 23 && 32 {
+  log("oh")
+}
```

Comment:
I'm not sure this case adds much value to the previous assertion. This seems more like specifics of the binary operator that we can improve on in the future.

Ideally, I'd just use an assertion that is inherently fallible and won't change anything in that regard in the future, e.g. `parse_json`.

Reply:
True. It was just the scenario that the fuzz testing kept bumping into. I've removed it now.